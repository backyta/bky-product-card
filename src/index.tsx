
export * from './components';
























//* Aqui se ponen todas las exportacion que van a ser vistas desde el mundo exterior, usualmente solo
//* se exponen los componentes es lo que exponemos.