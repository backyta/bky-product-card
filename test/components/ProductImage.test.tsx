import React from 'react'
import renderer from 'react-test-renderer'
import { ProductImage, ProductCard } from '../../src/components' 
import { product2 } from '../data/products';

describe('Pruebas en ProductImage', () => {
    
    test('debe de mostrar el componente correctamente con la imagen perzonalizada', () => {
        
        const wrapper = renderer.create(
            <ProductImage img='https://hla.jpg' className='Custom-class'/>
        )
        // console.log( wrapper.toJSON() );
        expect( wrapper.toJSON() ).toMatchSnapshot();
    });

    test('debe de mostrar el componete con la imagen del producto', () => {

        const wrapper = renderer.create(
            <ProductCard product={ product2 }>
                {
                    () =>(
                        <ProductImage />
                    )
                }
            </ProductCard>
        )

        expect( wrapper.toJSON() ).toMatchSnapshot();
    });
});