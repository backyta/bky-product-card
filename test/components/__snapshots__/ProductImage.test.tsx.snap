// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en ProductImage debe de mostrar el componente correctamente con la imagen perzonalizada 1`] = `
<img
  alt="Product Image"
  className="productImg Custom-class"
  src="https://hla.jpg"
/>
`;

exports[`Pruebas en ProductImage debe de mostrar el componete con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product Image"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
